engine/Assets/Scripts/Modes/Aether/ConnectToMultiplayerMode.cs

```diff
@@ -0,0 +1,167 @@
+using Google.Protobuf;
+using Synthesis.Runtime;
+using Synthesis.UI.Dynamic;
+using SynthesisAPI.Aether.Lobby;
+using SynthesisAPI.Controller;
+using SynthesisAPI.Utilities;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Logger = SynthesisAPI.Utilities.Logger;
+
+public class ConnectToMultiplayerMode : IMode {
+    private const string SERVER_IP   = "127.0.0.1";
+    private const string CLIENT_NAME = "Client 3.1415";
+    // clang-format off
+    private static readonly string[] _robotFolders = {
+        "Mira",
+        "Mira/Multiplayer"
+    };
+    // clang-format on
+    private string _multiplayerRobotFolder => _robotFolders[1];
+
+    private LobbyClient _client;
+
+    public ClientConnectionState ConnectionState { get; private set; } = ClientConnectionState.Disconnected;
+    public ClientActionState ActionState { get; private set; }         = ClientActionState.Idle;
+
+    public List<string> GetAvailableRobots() {
+        List<string> robots = new List<string>();
+
+        foreach (string folder in _robotFolders) {
+            string path = ParsePath(Path.Combine("$appdata/Autodesk/Synthesis", folder), '/');
+            if (!Directory.Exists(path)) {
+                Directory.CreateDirectory(path);
+            }
+
+            robots.AddRange(Directory.GetFiles(path).Where(x => Path.GetExtension(x).Equals(".mira")));
+        }
+
+        return robots;
+    }
+
+    public Task<Result<LobbyMessage?, Exception>> SelectRobot(string pathToRobotMira) {
+        if (!File.Exists(pathToRobotMira)) {
+            Logger.Log("Robot does not exist", LogLevel.Error);
+            return Task.FromResult(new Result<LobbyMessage?, Exception>(new Exception("Robot file does not exist.")));
+        }
+
+        var robot = new DataRobot {
+            Name = Path.GetFileNameWithoutExtension(pathToRobotMira),
+            Data = ByteString.CopyFrom(File.ReadAllBytes(pathToRobotMira)),
+            // Guid = _client.Guid
```

Comment:
I had a problem with casting from `ulong` from `ulong?` if someone wants to take a look.

Reply:
Underneath the assignment you can test if `_client.Guid.HasValue` and then assign `robot.Guid = _client.Guid.Value` if it does.